src/verifier/sev_snp_vtpm/mod.rs

```diff
@@ -0,0 +1,65 @@
+use super::{Attestation, TeeEvidenceParsedClaim, Verifier};
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use vtpm_snp::certs::{get_chain_from_amd, get_vcek_from_amd};
+use vtpm_snp::hcl::HclReportWithRuntimeData;
+use vtpm_snp::report::Validateable;
+use vtpm_snp::vtpm::{Quote, VerifyVTpmQuote};
+
+#[derive(Serialize, Deserialize)]
+struct VtpmSnpEvidence {
+    quote: Quote,
+    hcl_report: HclReportWithRuntimeData,
+}
```

Comment:
we should send over the `vcek_cert` from the CVM, and the CVM should fetch it from the IMDS. we should embed the `get_chain_from_amd` into the binary (or fetch once at runtime, or fetch from filesystem), since these are long-lived certs.

Reply:
we can use the vcek and ask but if the AS doesn't have its own ark, then we're not checking anything.

A fake TEE could just create it's own self-signed certificate chain and send that over, and we wouldn't know that it's not real. So we have to anchor in a root certificate that we either have or fetch independently.